### ISSUE TITLE
Cannot connect to MongoDB Atlas Cluster

### ISSUE THREAD

**Issue by AiusDa** [ArkerLabs/event-sourcing-nestjs#4]
I'm trying to connect to a MongoDB Atlas cluster.

**Implementation:** 

```
EventSourcingModule.forRoot({
      mongoURL: 'mongodb+srv://{user}:{password}@{cluster}.mongodb.net/{database}?retryWrites=true&w=majority'
})
```

**Expected behavior:**

Successful connection

**Current behavior**

`MongoNetworkError: failed to connect to server [{cluster}.mongodb.net:27017] on first connect [MongoNetworkError: connection 5 to {cluster}.mongodb.net:27017 closed`

**Environment**

Node version: v13.12.0
Yarn version: v1.22.4

package.json
```
{
  "version": "0.0.1",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@nestjs/common": "^7.0.0",
    "@nestjs/core": "^7.0.0",
    "@nestjs/cqrs": "^7.0.0",
    "@nestjs/platform-express": "^7.0.0",
    "@nestjs/swagger": "^4.5.10",
    "class-validator": "^0.12.2",
    "dotenv": "^8.2.0",
    "event-sourcing-nestjs": "^1.1.3",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^6.5.4"
  },
  "devDependencies": {
    "@nestjs/cli": "^7.0.0",
    "@nestjs/schematics": "^7.0.0",
    "@nestjs/testing": "^7.0.0",
    "@types/express": "^4.17.3",
    "@types/jest": "25.2.3",
    "@types/node": "^13.9.1",
    "@types/supertest": "^2.0.8",
    "@typescript-eslint/eslint-plugin": "3.0.2",
    "@typescript-eslint/parser": "3.0.2",
    "eslint": "7.1.0",
    "eslint-config-prettier": "^6.10.0",
    "eslint-plugin-import": "^2.20.1",
    "jest": "26.0.1",
    "prettier": "^1.19.1",
    "supertest": "^4.0.2",
    "ts-jest": "26.1.0",
    "ts-loader": "^6.2.1",
    "ts-node": "^8.6.2",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^3.7.4"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".spec.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}
```

**Comment by Nytyr** [comment]
Hello!

All the parameters in the Mongo URI are correct?

The Event Sourcing library internally invokes the official [mongodb driver](https://www.npmjs.com/package/mongodb) with the URI without changing it.

**Comment by AiusDa** [comment]
Hello @Nytyr, all params are correct, I've already test it on MongoDB Compass.

**Comment by Nytyr** [comment]
Can you try the URI in a minimal project with just the mongo driver and a js like this? 

So we can know if the problem is our library or the mongo one.

```

var MongoClient = require('mongodb').MongoClient;
var url = "mongodb+srv://{user}:{password}@{cluster}.mongodb.net/{database}?retryWrites=true&w=majority";

MongoClient.connect(url, function(err, db) {
  if (err) throw err;
  console.log("Success!");
  db.close();
});
```

Thanks.

**Comment by AiusDa** [comment]
Sure, in fact I could make it work using a local database without the Atlas cluster and it works fine:

`mongodb://root:root@localhost:27017`

For the Atlas cluster URL I couldn't find an option to configure `useNewUrlParser: true`.  I looked in the [documentation of eventstore](https://github.com/adrai/node-eventstore#provide-implementation-for-storage) but didn't find a configuration for that.

**Comment by AiusDa** [comment]
> Can you try the URI in a minimal project with just the mongo driver and a js like this?
> 
> So we can know if the problem is our library or the mongo one.
> 
> ```
> const MongoClient = require('mongodb').MongoClient;
> const uri = "mongodb+srv://{user}:{password}@{cluster}.mongodb.net/{database}?retryWrites=true&w=majority";
> const client = new MongoClient(uri, { useNewUrlParser: true });
> client.connect(err => {
>   if (err) throw err;
>   console.log("Success!");
>   client.close();
> });
> ```
> 
> Thanks.

It works:

```
(node:18398) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
Success!
```

**Comment by Nytyr** [comment]
Okay, maybe is because of the mongodb versión, because the event store library we're using uses the mongodb 2.1.X.

I will look into it.

**Comment by AiusDa** [comment]
Thanks @Nytyr.

**Comment by matheusgaya** [comment]
any news on this issue? I am having the same issue trying to connect to a Mongo Atlas cluster

**Comment by Nytyr** [comment]
Fixed in the 1.1.4 version 🎉 

Just add "&ssl=true" at the end of the URL.

mongodb+srv://{user}:{password}@{cluster}.mongodb.net/{database}?retryWrites=true&w=majority&ssl=true
